docker/serverless/env-start-esp.py

```diff
@@ -42,41 +38,72 @@ def assert_env_var(name):
         )
 
 
-assert_env_var("PORT")
-ARGS.append("--http_port={}".format(os.environ["PORT"]))
+def make_error_app(error_msg):
+    # error_msg must be a utf-8 or ascii bytestring
+    def error_app(environ, start_response):
+        start_response("503 Service Unavailable", [("Content-Type", "text/plain")])
+        return [error_msg, "\n"]
+
+    return error_app
+
+
+def serve_error_msg(error_msg):
+    print("Serving error handler with '{}'.".format(error_msg))
+    import wsgiref.simple_server
+
+    app = make_error_app(error_msg)
+    port = int(os.environ["PORT"])
+    server = wsgiref.simple_server.make_server("", port, app)
+    server.serve_forever()
 
-assert_env_var("ENDPOINTS_SERVICE_NAME")
-ARGS.append("--service={}".format(os.environ["ENDPOINTS_SERVICE_NAME"]))
 
-if "ENDPOINTS_SERVICE_VERSION" in os.environ:
-    ARGS.extend(
-        [
-            "--rollout_strategy=fixed",
-            "--version={}".format(os.environ["ENDPOINTS_SERVICE_VERSION"]),
-        ]
-    )
-else:
-    ARGS.append("--rollout_strategy=managed")
+def main():
+    CMD = "/usr/sbin/start_esp"
+
+    # The command being run must be the 0th arg.
+    ARGS = [CMD, "--enable_backend_routing"]
+
+    assert_env_var("PORT")
```

Comment:
why this one is not wrapped with try/catch block?

Reply:
Because if there is no PORT in the environ, we can't serve an error on that port. I will add a comment.